### PR TITLE
docs: changelog + session log for 2026-08-03

### PR DESCRIPTION
**Pull request by Ravenwater**
## Summary

Day wrapup: `CHANGELOG.md` under `[Unreleased]`, plus `docs/sessions/2026-08-03-complex-factorizations-and-silent-acceptance.md`.

Covers everything merged after the v5.8.0 wrapup commit:

| PR | |
|---|---|
| #348 | team-parallel B packing — GEMM efficiency 79% → 91% |
| #350 | refreshed scaling data and results page |
| #354 | the PPE lab |
| #356 | `ldlt` Hermitian fix (#352) |
| #358 | `col_major` sparse container guards (#355) |
| #360 | Hermitian Cholesky (#353) |
| #361 | complex Householder / QR / LQ (#353) |

Also: #110 closed on measurement, #357 consolidated into #353, #362 filed.

## What the session log is actually for

The through-line was not the subject matter. It was one failure shape met five times:

> Something reports success while doing the wrong thing — and the test that should have caught it passes, because it is structurally aimed elsewhere.

Four instances were in the library. **One was in the guard written to prevent it**: the #361 complex guard went into `eigenvalue_symmetric_generic`, which does not call `householder` at all, while `eigen_symmetric` — the function performing the reduction — was left open. Its compile-failure test matched the regex and passed while covering nothing of the hazard.

That is written up plainly rather than smoothed over, because the harness it defeated had been introduced the same week and cited twice that day as the answer to exactly this problem. The transferable rule is in the log: *a negative control on the guard proves nothing unless you have also checked the guard is on the function that matters.*

The log also records **what did not get built**. #110 offered three implementation options; an `mc` sweep showed `T=1` throughput falls monotonically with `mc`, so the region all three would unlock is the region that measures worse — a measured zero. Worth keeping, since the issue had already been re-diagnosed twice on reasoning rather than measurement.

## Notes

- Session logs are not in the docs-site `FILE_MAP` (checked) — repo-only, no site changes needed.
- Docs-only; no code touched.

Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Documentation**
  * Added release notes covering complex Hermitian factorizations, Householder transformations, QR/LQ workflows, and related restrictions.
  * Documented Hermitian LDLT fixes, sparse-container validation, edge-case handling, and GEMM performance improvements.
  * Added a detailed engineering session report with benchmark findings, compatibility notes, validation outcomes, and follow-up items.

* **Tests**
  * Documented compile-failure coverage, diagnostic matching, reconstruction checks, and safeguards for unsupported configurations.

### CODE REVIEW

File: docs/sessions/2026-08-03-complex-factorizations-and-silent-acceptance.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  for cache reasons and parallelism starves it — is false here. Growing `mc` is
+  worse with no threading involved at all, so the region all three options would
+  unlock is the region that measures badly.
+- **The coupling is real but harmless.** The two imbalanced points collapse
+  exactly as the issue predicts, while every balanced point sits at 86–91%. The
+  mechanism was right; the inference that it cost anything was not.
+
+The acceptance bar (≥85%) had already been met by #348 the same day. Closed.
+**Had the issue's plan been implemented as written, it would have been days of
+work for a measured zero.**
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply docs/sessions/2026-08-03-complex-factorizations-and-silent-acceptance.md:149]: **Confirmed and fixed** in 33d6a1b. You are right, and this one mattered more than a wording nit because I used it as the evidence for closing #110.

The measured column:

| mc | 32 | 64 | 128 | 192 | 256 | 320 |
|---|---:|---:|---:|---:|---:|---:|
| T=1 | 57.32 | 56.81 | 54.84 | 53.71 | **47.89** | **48.80** |

It falls through `mc=256` and rises at `mc=320`. The cited range in the #110 comment (32→256) *is* monotonic, which is how the word survived drafting — but the table sits directly above the sentence and includes the rebound, so the claim reads as contradicting its own data.

I did not take the proposed wording ("generally falls, with a small rebound"), because it keeps the emphasis on the curve's shape and the shape was never the argument. The claim that closes #110 is:

> no `mc` larger than the shipping 64 is better at `T=1`

which is what the three candidate options would need in order to pay off — they all exist to let `mc` grow. So the log now leads with that, and states the rebound as a measured fact rather than a caveat: 48.80 at `mc=320` is still **14% below** `mc=64`, and no point above 64 comes within 13%. Stronger claim, and one the data actually licenses.

The superseded wording is noted in place rather than silently swapped, and **#110 carries a public correction** ([comment](https://github.com/stillwater-sc/mtl5/issues/110#issuecomment-5178271783)) since the claim was published there, not just here.

Worth naming why this is the right catch on this particular issue: #110 was re-diagnosed twice today on reasoning rather than measurement — once from blocking parameters read out of the wrong build configuration, once from a mechanism assumed costly without checking whether it was active. Closing it on a word stronger than the measurement supports would have been the third instance of the same habit, in the write-up of the very session that documented it.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,31 @@ Format follows [Conventional Commits](https://www.conventionalcommits.org/).
 
 ## [Unreleased]
 
+### Added
+
+#### Complex (Hermitian) dense factorizations (#353)
+`cholesky`, `qr` and `lq` did not compile for `std::complex`, failing first on relational operators applied to a complex where a *magnitude* was meant. Repairing only those comparisons would have produced routines that compile and compute the **wrong** factorization — the unconjugated inner product yields `L·Lᵀ`, not `L·Lᴴ` — so neither fix repairs a comparison.
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply CHANGELOG.md]: **Confirmed and fixed** in 33d6a1b. The paragraph did attribute a Cholesky-specific mechanism to all three routines.

I took a slightly different wording than proposed, because "QR and LQ require the complex Householder path described below" points at the fix rather than naming the failure, and the failure is the part worth recording — the whole reason neither PR repairs a comparison is that the comparisons were never the substance.

Now:

> Repairing only those comparisons would have produced routines that compile and compute the **wrong** result, by a different mechanism in each: `cholesky`'s unconjugated inner product yields `L·Lᵀ` rather than the `L·Lᴴ` a Hermitian matrix has, while `qr`/`lq` additionally need `Σ|z|²` in place of `Σz²` and a reflector written `I − τ·v·vᴴ` rather than `I − β·v·vᵀ`.

Both mechanisms named, and "additionally" keeps the shared part (an unconjugated product) visible while separating what is specific to each.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,31 @@ Format follows [Conventional Commits](https://www.conventionalcommits.org/).
...
+- **`householder` broke its own real-β guarantee** for a complex vector with a vanishing tail and a complex leading entry, returning `tau = 0` and leaving a complex `R`/`L` diagonal — silent, since `Q` stayed unitary and `Q·R` still reproduced `A`. The identity shortcut is now taken only when it is exact (#361)
+
+### Performance
+- **Blocked GEMM parallel efficiency 79% → 91%** on 8 physical cores. The shared B panel was packed by each jc-team's leader while the rest of the team waited on the barrier — a serial region proportional to `kc·nc` per `pc` step (at N=2048 with a team of 8, one thread packed ~67 MB while seven idled). The team now splits the NR-column panels; packing is pure data movement into disjoint offsets, so the packed bytes and the result are unchanged (#348)
+
+### Changed
+- `docs/sparse-direct-solvers-design.md` no longer advertises a "zero-cost CSC view over `compressed2D` with col-major parameters" — never implemented; `sparse/util/csc.hpp` uses a standalone owning `csc_matrix`. `docs/architecture/aggregate-types.md` no longer claims "Full support (CSR, CSC, COO, ELLPACK)"; CSC exists only as a conversion utility (#358)
```
[the remark this answers is not in the document]
Comment (Ravenwater) [reply CHANGELOG.md:31]: **Not taking this one.** #110's closure, #357's consolidation and #362's filing are issue-triage outcomes, not changes to the library, and `CHANGELOG.md` documents what changed for someone consuming MTL5.

Concretely, none of the three has anything a consumer could act on:

- **#110** closed with **zero code change**. The efficiency it was about had already been delivered by #348, which *is* in the changelog, under Performance. A line saying "#110 was closed" would tell a reader nothing that the #348 entry does not already tell them, while implying something shipped.
- **#357** was a near-duplicate I filed and then consolidated into #353. That is a mistake of mine being cleaned up. #353's outcome is in the changelog as the Cholesky and QR/LQ entries.
- **#362** is work that does **not** exist yet. Listing unfiled-but-planned capability under `[Unreleased]` inverts what that heading means — `[Unreleased]` is "merged, not yet tagged", not "intended".

On the suggested alternative — stating explicitly that these are session-log-only — I would rather not. A changelog carrying a meta-note about its own exclusions is a small ongoing maintenance burden and an invitation to argue the boundary on every entry. The boundary is the conventional one and does not need restating in the artifact.

The session log is the right home and already has all three, with the reasoning attached: #110 under the measurement that closed it, #357 under process notes as a filing mistake, #362 under follow-ups. That split is deliberate — the changelog says what changed, the session log says what happened.

One thing your comment did surface that I have acted on: `#362` was not linked from the changelog at all, so a reader of the `hessenberg`/`eigen_symmetric` rejection had no pointer to why the gap exists or where it is tracked. That entry now names it, which is a genuine consumer-facing need and different from a triage line.

[the remark this answers is not in the document]
Comment (Ravenwater) [reply CHANGELOG.md:31]: Correction to my own reply: the last paragraph is wrong. **#362 was already referenced** in the changelog in the original commit — `git show b02b2ec:CHANGELOG.md` has it on the `hessenberg`/`eigen_symmetric` line. I claimed to have acted on something that was already there, and took credit for a change I did not make.

The substance of the decline is unchanged: triage outcomes stay in the session log, changes stay in the changelog, and the one consumer-facing pointer that mattered was present from the start.
